.github/workflows/test-linux.yml

```diff
@@ -67,8 +64,56 @@ jobs:
         uses: ./.github/workflows/test
         with:
             name: 'unit'
-            filter: 'FullyQualifiedName!~Elastic.Apm.StartupHook.Tests&FullyQualifiedName!~Elastic.Apm.Profiler.Managed.Tests'
-  
+            filter: 'FullyQualifiedName!~Elastic.Apm.StartupHook.Tests&FullyQualifiedName!~Elastic.Apm.Profiler.Managed.Tests&FullyQualifiedName!~Elastic.Apm.Azure'
+            
+  azure-tests:
+    runs-on: ubuntu-latest
+    if: |
+      github.event_name != 'pull_request'
+      || github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork == false
+    steps:
+      - uses: actions/checkout@v3
+      - name: Bootstrap Action Workspace
+        uses: ./.github/workflows/bootstrap
+        with:
+          azure: 'true'
+
+      - name: Build
+        shell: bash
+        run: dotnet build -c Release --verbosity minimal
+
+      - name: 'Read credentials'
+        uses: hashicorp/vault-action@v2.7.1
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          method: approle
+          secrets: |
+            secret/apm-team/ci/apm-agent-dotnet-azure client_id | ARM_CLIENT_ID ;
+            secret/apm-team/ci/apm-agent-dotnet-azure client_secret | ARM_CLIENT_SECRET ;
+            secret/apm-team/ci/apm-agent-dotnet-azure subscription_id | ARM_SUBSCRIPTION_ID ;
+            secret/apm-team/ci/apm-agent-dotnet-azure tenant_id | ARM_TENANT_ID
```

Comment:
This will work only on feature branches, secrets are not accessible from forks 

cc @elastic/ci-robots